# 3. Создайте собственный класс-исключение, который должен проверять содержимое списка
# на наличие только чисел. Проверить работу исключения на реальном примере.
# Запрашивать у пользователя данные и заполнять список необходимо только числами.
# Класс-исключение должен контролировать типы данных элементов списка.
#
# Примечание: длина списка не фиксирована. Элементы запрашиваются бесконечно,
# пока пользователь сам не остановит работу скрипта, введя, например, команду «stop».
# При этом скрипт завершается, сформированный список с числами выводится на экран.
#
# Подсказка: для этого задания примем, что пользователь может вводить только числа и строки.
# Во время ввода пользователем очередного элемента необходимо реализовать проверку типа элемента.
# Вносить его в список, только если введено число. Класс-исключение должен не позволить
# пользователю ввести текст (не число) и отобразить соответствующее сообщение.
# При этом работа скрипта не должна завершаться.


class OwnError(Exception):
    def __init__(self, txt):
        self.txt = txt


''' Определение итогового списка'''
my_list = []
while True:
    try:
        inp_data = input("Введите положительное число (для завершения введите stop ): ")
        el = list(inp_data)

        '''Обработка окончания ввода'''
        if inp_data == 'stop':
            print("Ввод чисел завершен")
            print(f" Итоговый список чисел: {my_list}")
            break
        ''' Введение логической переменнй OwnError – исключения '''
        T = False
        if ord(el[0]) in [45, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57] and el.count('.') < 2 and el[len(el) - 1] != '.':
            T = True
        if len(el) > 1 and ord(el[0]) == 45:
            if ord(el[1]) != 46:
                T *= True
            else:
                T *= False
        if len(el) == 1 and ord(el[0]) == 45:
            T *= False
        for i in range(1, len(el)):
            if ord(el[i]) in [46, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57]:
                T *= True
            else:
                T *= False
        if T == 0:
            raise OwnError("Вы ввели не число, введите число еще раз")
    except OwnError as err:
        print(err)
    except IndexError:
        print("Вы ввели не число")
    else:
        print(f"Все хорошо. Ваше число: {inp_data}")
        my_list.append(inp_data)
        continue
